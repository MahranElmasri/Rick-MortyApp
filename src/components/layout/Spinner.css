.spinner {
  width: 200px;
  height: 200px;
  opacity: 0.5;
  margin: auto;
  display: block;
}
