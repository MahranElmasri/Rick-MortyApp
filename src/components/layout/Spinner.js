import "./Spinner.css";
import React from "react";
import spinner from "./spinner.gif";

export default function Spinner() {
  return (
    <div data-testid="spinner-test" className="spinner">
      <img data-testid="img" src={spinner} alt="Loading" />
    </div>
  );
}
