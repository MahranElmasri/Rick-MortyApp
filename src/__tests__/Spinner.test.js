import React from "react";

import Spinner from "../components/layout/Spinner";

afterEach(cleanup);

it("should render correctly", () => {
  const { asFragment } = render(<Spinner />);
  expect(asFragment()).toMatchSnapshot();
});

it("should test classname", () => {
  const { getByTestId, getByText } = render(<Spinner />);
  expect(getByTestId("spinner-test")).toHaveClass("spinner");
});
