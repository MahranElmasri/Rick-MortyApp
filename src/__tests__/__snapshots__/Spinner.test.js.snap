// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div
    class="spinner"
    data-testid="spinner-test"
  >
    <img
      alt="Loading"
      data-testid="img"
      src="spinner.gif"
    />
  </div>
</DocumentFragment>
`;
